pkg/controllers/dynakube/extension/tls/reconciler.go

```diff
@@ -52,13 +59,18 @@ func (r *reconciler) reconcileSelfSigned(ctx context.Context) error {
 	})
 
 	if err != nil && k8serrors.IsNotFound(err) {
-		return r.createSelfSignedTLSSecret(ctx)
+		err = r.createSelfSignedTLSSecret(ctx)
+		if err != nil {
+			return err
+		}
+
+		conditions.SetSecretCreated(r.dk.Conditions(), extensionsSelfSignedTLSSecretConditionType, getSelfSignedTLSSecretName(r.dk.Name))
```

Comment:
Shouldn't  the condition also be set accordingly if creating the certificate failed for whatever reason?

Reply:
True :+1: My brain is still coming from vacations. Im updating it